pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -136,23 +137,25 @@ def get_model_from_plugin(
         Returns: Reference :class:`LightningModule` within parallel wrapper.
 
         """
-        if isinstance(model, LightningDistributedDataParallel):
-            return model.module
+        if isinstance(model, DistributedDataParallel):
+            model = model.module
+        if isinstance(model, LightningDistributedWrapper):
+            model = model.module
         return model
 
     @contextmanager
-    def block_backward_sync(self, model: LightningDistributedDataParallel):
+    def block_backward_sync(self, model: DistributedDataParallel):
         """
         Blocks ddp sync gradients behaviour on backwards pass.
         This is useful for skipping sync when accumulating gradients, reducing communication overhead
         Returns: context manager with sync behaviour off
         """
         yield model.no_sync()
```

Comment:
do we need `block_backward_sync` still? can we directly call `model.no_sync()`

Reply:
the only reason I can think of why it is there is so the user can override this method in their own plugin, though there is not much customization they can do to this context manager :)

Reply:
We basically added this, since we did not want anything that is only DDP specific (i.e. any typechecks against the prior `LightningDistributedDataParallel` within the trainer/training-loop as this one should be backend agnostic.